xbmc/network/DNSNameCache.cpp

```diff
@@ -84,20 +109,42 @@ bool CDNSNameCache::Lookup(const std::string& strHostName, std::string& strIpAdd
 #endif
 
   // perform dns lookup
-  struct hostent *host = gethostbyname(strHostName.c_str());
-  if (host && host->h_addr_list[0])
+  struct addrinfo hints;
+  struct addrinfo *result(NULL);
+
+  memset(&hints, 0, sizeof(struct addrinfo));
+  int err;
+
+  // prefer DNS record type (A vs AAAA)  be this of active interface(address)
+  // otherwise (by default) system prefers A(IPv4) records and this will make
+  // troubles on dual configured hosts - but without wide area access over IPv4
+  do {
+    freeaddrinfo(result);
```

Comment:
On the first loop this is called with result beeing a NULL ptr - i don't see it mentioned anywhere that calling this function with NULL is safe. You got any sources that this is a safe thing to do?